File: frontend/src/types/chemical.ts
export interface Chemical {
  id: string;
  dtxsid: string | null;
  cas_number: string | null;
  preferred_name: string;
  inci_name: string | null;
  molecular_formula: string | null;
  molecular_weight: number | null;
  smiles: string | null;
  description: string | null;
  hazard_data: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface RegulatoryStatus {
  id: string;
  chemical_id: string;
  list_name: string;
  list_full_name: string;
  status: string;
  classification: string | null;
  details: Record<string, unknown>;
  effective_date: string | null;
  source_url: string | null;
}

File: backend/app/models/chemical.py
from pydantic import BaseModel


class Chemical(BaseModel):
    id: str
    dtxsid: str | None = None
    cas_number: str | None = None
    preferred_name: str
    inci_name: str | None = None
    molecular_formula: str | None = None
    molecular_weight: float | None = None
    description: str | None = None
    hazard_data: dict = {}


class RegulatoryStatus(BaseModel):
    list_name: str
    status: str | None = None
    classification: str | None = None
    details: dict = {}
